workers/src/codec/bytes_utils.rs

```diff
@@ -0,0 +1,49 @@
+pub trait BytesUtils {
+    fn to_u8(&self, start: usize) -> u8;
+    fn to_u16(&self, start: usize) -> u16;
+    fn to_u32(&self, start: usize) -> u32;
+    fn to_u64(&self, start: usize) -> u64;
+    fn to_u128(&self, start: usize) -> u128;
+    fn to_bytes32(&self, start: usize) -> &[u8];
+    fn to_byte_array<const N: usize>(&self, start: usize) -> [u8; N];
+}
+
+impl BytesUtils for &[u8] {
```

Comment:
This is error prone and inefficient too, we shouldn't be copying this. `bytes::BytesMut` from crates.io avoids this handrolled pointer slicing